provider/views.py

```diff
@@ -255,7 +325,36 @@ def error_response(self, request, error, **kwargs):
 
         return self.render_to_response(ctx, **kwargs)
 
+    def get_implicit_response(self, request, client):
+        """
+        Assumes validated and authorized request with a response_type that is
+        associated with an implicit flow.
+        """
+        data = self.get_data(request)
+
+        lookup_kwargs = {
+            "user": request.user,
+            "client": client,
+            "scope": scope.to_int(*data.get('scope', '').split())
+        }
+
+        if constants.SINGLE_ACCESS_TOKEN:
+            token = self.get_access_token(request, **lookup_kwargs)
+        else:
+            token = self.create_access_token(request, **lookup_kwargs)
+
+        response_data = self.access_token_response_data(token, data['response_type'])
+
+        basepath = data.get("redirect_uri")
+        if not basepath:
+            basepath = token.client.redirect_uri
```

Comment:
According to the spec, the `redirect_ui` should be validated:
```
The authorization server validates the request to ensure that all
   required parameters are present and valid.  The authorization server
   MUST verify that the redirection URI to which it will redirect the
   access token matches a redirection URI registered by the client as
   described in Section 3.1.2.
```

I recommend we require that the redirect_uri in the request (if present) match the previously registered one that is associated with the `Client` in the database.